crates/noirc_evaluator/src/ssa_refactor/acir_gen/acir_ir/generated_acir.rs

```diff
@@ -390,32 +390,42 @@ impl GeneratedAcir {
             "range check with bit size of the prime field is not implemented yet"
         );
 
-        let mut aof = a + offset;
+        let mut lhs_offset = lhs + offset;
 
-        if b.is_const() && b.q_c.fits_in_u128() {
-            let f = if *offset == Expression::one() {
-                aof = a.clone();
-                assert!(b.q_c.to_u128() >= 1);
-                b.q_c.to_u128() - 1
+        // Optimisation when rhs is const
+        if rhs.is_const() && rhs.q_c.fits_in_u128() {
+            // We try to move the offset to rhs
+            let rhs_offset = if *offset == Expression::one() {
+                lhs_offset = lhs.clone();
+                assert!(rhs.q_c.to_u128() >= 1);
```

Comment:
can you add a panic message to this assert, as to why this should not happen?

Reply:
This was not supposed to happen because this case was optimised out. I am not so sure anymore, so I made it work even if the assert is false.